libkbfs/folder_branch_ops.go

```diff
@@ -3449,7 +3470,11 @@ func (fbo *folderBranchOps) Rename(
 			getNodeIDStr(newParent), newName, err)
 	}()
 
-	err = fbo.checkNode(newParent)
+	err = fbo.checkNodeForWrite(ctx, oldParent)
+	if err != nil {
+		return err
+	}
+	err = fbo.checkNodeForWrite(ctx, newParent)
```

Comment:
I'm wondering if you need a lock between these two.

Reply:
I don't believe any lock is needed here.  The `Readonly()` method only takes a context as input, so any locking in FBO should be independent of what it decides.  Let me know if I'm missing something though.

Reply:
Can the checks be invalidated before the lock is taken?

Reply:
I guess one could build a `Node` implementation that does so, but even if you did, taking a lock in FBO wouldn't be able to prevent it in all cases, only in the case where the catalysis for change was coming through FBO itself somehow.  You can imagine systems like that, but it doesn't seem worth engineering a fix for, since it wouldn't fix the bigger problem anyway.

I could add a second check behind all the locks if you feel strongly about it, it's probably fairly cheap, just a bit of extra code.

Reply:
Naw, I think it's extremely low risk that this would ever come up. I just saw 2 operations on mutable data not under lock and red flags went off.

Reply:
Yeah good instinct.  Though in this case, a race wouldn't be just limited to this double-check, but to any check that happens before the lock (i.e., all of them).  But yeah, agreed it's not worth worrying about now.